mono/utils/mono-error.c

```diff
@@ -648,7 +660,24 @@ mono_error_prepare_exception (MonoError *oerror, MonoError *error_out)
 				break;
 			}
 
-			exception = mono_exception_from_name_two_strings_checked (mono_defaults.corlib, "System", "MissingMethodException", type_name, method_name, error_out);
+			MonoString *signature = NULL;
+			if (error->member_signature) {
+				signature = string_new_cleanup (domain, error->member_signature);
+				if (!signature) {
+					mono_error_set_out_of_memory (error_out, "Could not allocate signature");
+					break;
+				}
+			}
+
+			MonoString *message = NULL;
+			if (error->full_message && strlen (error->full_message) > 0) {
+				message = string_new_cleanup (domain, error->full_message);
+				if (!message) {
+					mono_error_set_out_of_memory (error_out, "Could not allocate message");
+					break;
+				}
+			}
+			exception = mono_exception_from_name_four_strings_checked (mono_defaults.corlib, "System", "MissingMethodException", type_name, method_name, signature, message, error_out);
```

Comment:
You also need to update https://github.com/mono/mono/blob/master/mcs/class/corlib/LinkerDescriptor/mscorlib.xml#L252